src/Common/src/TypeSystem/Interop/IL/MarshalHelpers.cs

```diff
@@ -69,6 +70,12 @@ public static bool UseLazyResolution(MethodDesc method, string importModule, PIn
             if (forceLazyResolution.HasValue)
                 return forceLazyResolution.Value;
 
+            string assemblySimpleName = ((EcmaAssembly)((MetadataType)method.OwningType).Module).GetName().Name;
```

Comment:
Can you cast to `IAssemblyDesc` to avoid ECMA stuff in this file? Also, a comment why we need this hack would be nice.